Sprint 2/enemy attack

roosa2000 opened the pull request:
<!---
Pull request template for 603 Game 3. Feel free to remove comments as you fill out information.
Original template by Rob Reddick
--->
## Summarize what is being added
<!--- Can use bullet lists here to cover new additions, but give a bit more detail than what is given in commit messages --->

--> Implemented Player Detection in a Conical field of View Angle
--> Enemy is able to attack the player when the player is in conical FOV
--> Enemy will not shoot continuously, there is a cooldown time when the player is in range
--> Enemy is able to look in the direction of the player
--> Enemy's weapon also rotates towards the enemy in y axis
--> Player will rotates horizontally as per the mouse position
--> Player can aim the weapon by rotating it vertically using the mouse

## Please describe how to test
<!---Give the important steps needed for testing your main changes--->

--> In the scenes folder open the test folder -> enemy scene.

- Play the scene.

- Move the player around.

- The enemy would rotate and shoot towards the player only when the player is in front of the enemy and within the conical detection range.

- Check if the cooldown is working as expected.

- It should not be rotating, attacking or detecting the player if the player is behind or not in the conical field of view angle.

- To test if the player is in the range please follow the debug logs

- Use move the mouse horizontally to rotate the player and vertically to rotate  or aim the weapon

## Issue worked on
<!---Paste in a link to the issue this PR addresses--->
https://timeisup.atlassian.net/browse/SCRUM-18?atlOrigin=eyJpIjoiOTI0OTE4MDA5MGRmNDkxMDg5MjRjY2UyNzUyYzRlMGMiLCJwIjoiaiJ9

https://timeisup.atlassian.net/browse/SCRUM-15?atlOrigin=eyJpIjoiZjQ2ZWNiMDFhYmEzNDQ3Mjg5NjZjZGFlNGE4ZDVlMTgiLCJwIjoiaiJ9

## What Sprint is this due in?
<!---Only need to indicate here which week/deliverable the PR is for--->
Sprint 2

Commit:
Added Enemy to the scene

Commit:
Added Enemy to the scene

Commit:
Enemy is able to detect player

Commit:
Enemy is able to detect player

roosa2000 committed:
Implemented Conical field of view angle for the enemy

roosa2000 committed:
Implemented Conical field of view angle for the enemy

RohitK-RIT committed:
Turned active input handling to both

RohitK-RIT reviewed (changes requested):
Your PR has a few issues:

- [x] Merges into the main, should merge in sprint 2.
- [x] Rebase your code on the sprint 2 branch.
- [x] Structural changes (as discussed)

heenathadani committed:
added basic stats

Commit:
Implemented Physics.OverlapSphere for detecting players in the range and storing it in a list

Commit:
Implemented Physics.OverlapSphere for detecting players in the range and storing it in a list

Commit:
Implemented Player to look in the direction of mouse cursor

Commit:
Implemented Player to look in the direction of mouse cursor

Commit:
Added AIInputHsndler class

Commit:
Added AIInputHsndler class

heenathadani committed:
updated char stats

heenathadani committed:
bug fixes

heenathadani committed:
completed char stats

heenathadani committed:
fixed naming

Commit:
Implemented Basic Enemy Attack

Commit:
Implemented Basic Enemy Attack

heenathadani committed:
updated stats

heenathadani committed:
added player hud

heenathadani committed:
added functional player hud

heenathadani committed:
Merge branch 'Sprints/Sprint-2' into CharacterStats

Commit:
Fixed the continuous shooting issue

Commit:
Fixed the continuous shooting issue

Commit:
Implemented mouse position using new input system and fixed rotation issue

Commit:
Implemented mouse position using new input system and fixed rotation issue

Commit:
Fixed Merge conflict changes

Commit:
Fixed Merge conflict changes

heenathadani committed:
updated player hud with no playercontroller ref

heenathadani committed:
Merge branch 'Sprints/Sprint-2' into CharacterStats

heenathadani committed:
added health bar functionality

heenathadani committed:
full health

heenathadani committed:
bug fixessss

Commit:
Changed the code structure

Commit:
Changed the code structure

Commit:
File changes

Commit:
File changes

Commit:
Added method for enemy rotation

Commit:
Added method for enemy rotation

RohitK-RIT committed:
Look Input implemented for player and enemy

RohitK-RIT committed:
Look Input implemented for player and enemy

RohitK-RIT committed:
Minor changes

Commit:
Added player tag to player prefab and added comments

Commit:
Added player tag to player prefab and added comments

RohitK-RIT committed:
Added MaxAmmo implementation

RohitK-RIT committed:
Fixed ranged weapon shooting after ammo over issue.

Commit:
Enemy is now able to aim the player in the correct angle

Commit:
Enemy is now able to aim the player in the correct angle

Commit:
Fixed conical FOV rotation issue

Commit:
Fixed conical FOV rotation issue

RohitK-RIT committed:
Merge pull request #12 from RohitK-RIT/CharacterStats

Character stats

RohitK-RIT committed:
Added Materials to the different characters
Replaced the Player prefab in the Assets\_Project\Scenes\Test Scenes\CharStatsTest.unity

Merge branch 'Sprint_1/Enemy-Attack' of https://github.com/RohitK-RIT/Flux-TimesUp into Sprint_1/Enemy-Attack

# Conflicts:
#	Assets/_Project/Scripts/Core/Player Controllers/LocalPlayerController.cs
#	Assets/_Project/Scripts/Core/Player Controllers/PlayerController.cs

Commit:
Code Clean up

RohitK-RIT committed:
Removed Debug.Log for bullet count.

RohitK-RIT committed:
Clamped pitch angles for look direction in the movement controller to +90 and -90.

RohitK-RIT reviewed (dismissed):
Clamped the weapon pitch angle
